File: eel/plugins/models/kisstomato_model/templates/javascript.js
/*
	{{ o.desc }}
*/

// kisstomato-js-{{ o.name }}-a-start-user-code-kisstomato
// kisstomato-js-{{ o.name }}-a-stop-user-code-kisstomato

&&  for oSection in o.sections
// --------------------------------------------------------------------------
// Section : {{ oSection.name }}
/* {{ oSection.desc }} */
// --------------------------------------------------------------------------

// kisstomato-section-{{ oSection.name }}-start-user-code-kisstomato
// kisstomato-section-{{ oSection.name }}-stop-user-code-kisstomato

&&  endfor

&&  for oFonction in o.fonctions
/* {{ oFonction.desc }} */
&&      if oFonction.args|length > 0
// Argument{% if oFonction.args|length > 1 %}s{% endif %} :
&&          for oArg in oFonction.args
// - {{ oArg[ 'name' ] }} : {{ oArg[ 'type' ] }} : {% if oArg[ 'require' ] %}(obligatoire) {% endif %}{{ oArg[ 'desc' ] }}
&&          endfor
&&      endif
function {{ oFonction.name }}({% for oArg in oFonction.args %}{% if oArg != oFonction.args[ 0 ] %}, {% endif %}{{ oArg[ 'name' ] }}{% endfor %}){
&&          if oFonction[ 'return-val' ]
	let oResult = null;

&&          endif

	// kisstomato-function-{{ oFonction.name }}-start-user-code-kisstomato
	// kisstomato-function-{{ oFonction.name }}-stop-user-code-kisstomato

&&          if oFonction[ 'return-val' ]
	return oResult;
&&          endif
}

&&  endfor

&&  for oR in o.jquerys_ready
/* {{ oR.desc }} */
$( document ).ready( function(){
	// kisstomato-ready-{{ oR.name }}-start-user-code-kisstomato
	// kisstomato-ready-{{ oR.name }}-stop-user-code-kisstomato
} );

&&  endfor

&&  for oO in o.jquerys_on
/* {{ oO.desc }} */
$( document ).on( "{{ oO.event }}", "{{ oO.selector }}", function() {
	// kisstomato-on-{{ oO.name }}-start-user-code-kisstomato
	// kisstomato-on-{{ oO.name }}-stop-user-code-kisstomato
} );

&&  endfor

// kisstomato-js-{{ o.name }}-b-start-user-code-kisstomato
// kisstomato-js-{{ o.name }}-b-stop-user-code-kisstomato

File: eel (copie)/plugins/models/kisstomato_model/classes/javascript.py
from core import generator, nodeElement

class javascript(nodeElement):
    oRoots = []
    def __init__(self, oJs):
        oItems = self._getItems( oJs )
        self.name = oJs[ 'text' ]
        self.desc = oItems[ 'desc' ][ 'value' ]

        # recupere les sections
        self.sections = []
        for oSection in generator.getNodesByTypes( oJs, 'javascript/js_sections/section' ):
            oItems = self._getItems( oSection )
            self.sections.append( { 'name': oSection[ 'text' ], 'desc': oItems[ 'desc' ][ 'value' ] } )

        # recupere les fonctions
        self.fonctions = []
        for oFonction in generator.getNodesByTypes( oJs, 'javascript/fonctions/fonction' ):
            oItems = self._getItems( oFonction )

            # recupere les arguments
            oArgs = []
            for oArg in generator.getNodesByTypes( oFonction, 'fonction/arguments/argument' ):
                oItemsA = self._getItems( oArg )
                oArgs.append( { 'name': oArg[ 'text' ], 'desc': oItemsA[ 'desc' ][ 'value' ], 'require': oItemsA[ 'require' ][ 'value' ], 'type': oItemsA[ 'type' ][ 'value' ] } )

            self.fonctions.append( { 'name': oFonction[ 'text' ], 'desc': oItems[ 'desc' ][ 'value' ], 'args': oArgs, 'return-val': oItems[ 'return-val' ][ 'value' ] } )

        # recupere les jquery_ready
        self.jquerys_ready = []
        for oJQR in generator.getNodesByTypes( oJs, 'javascript/jquerys/jquery_ready' ):
            oItems = self._getItems( oJQR )
            self.jquerys_ready.append( { 'name': oJQR[ 'text' ], 'desc': oItems[ 'desc' ][ 'value' ] } )

        # recupere les jquery_on
        self.jquerys_on= []
        for oJQO in generator.getNodesByTypes( oJs, 'javascript/jquerys/jquery_on' ):
            oItems = self._getItems( oJQO )
            self.jquerys_on.append( { 'name': oJQO[ 'text' ], 'desc': oItems[ 'desc' ][ 'value' ], 'event': oItems[ 'event' ][ 'value' ], 'selector': oItems[ 'selector' ][ 'value' ] } )
